primitives/state-machine/src/overlayed_changes/changeset.rs

```diff
@@ -0,0 +1,635 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2017-2020 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License
+
+//! Houses the code that implements the transactional overlay storage.
+
+use super::{StorageKey, StorageValue};
+
+use itertools::Itertools;
+use std::collections::{HashSet, BTreeMap, BTreeSet};
+use smallvec::SmallVec;
+
+const PROOF_DIRTY_KEYS: &str = "\
+	We assume transactions are balanced. Every start of a transaction created one dirty
+	keys element. This function is only called on transaction close. Therefore an element
+	created when starting the transaction must exist; qed";
+
+const PROOF_DIRTY_OVERLAY_VALUE: &str = "\
```

Comment:
Same here, this proof is used only once.

Reply:
Inlined.